src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Sdk.DefaultItems.targets

```diff
@@ -13,6 +13,14 @@ Copyright (c) .NET Foundation. All rights reserved.
 
   <!-- Don't get the default item globs by default when the flag is not set. -->
   <PropertyGroup Condition="'$(UsingNETSdkDefaults)' == 'true'">
+    <!-- Given DisableDefaultItemsInProjectFolder=true, set EnableDefaultItems=false
+         in case there is a solution/project in the current directory. -->
+    <EnableDefaultItems Condition="'$(EnableDefaultItems)' == '' And
+      '$(DisableDefaultItemsInProjectFolder)' == 'true' And
+      ('$([System.IO.Directory]::GetFiles($(MSBuildProjectDirectory), &quot;*.sln&quot;))' != '' Or
+      $([System.IO.Directory]::GetFiles($(MSBuildProjectDirectory), &quot;*.slnx&quot;)) != '' Or
+      $([System.IO.Directory]::GetFiles($(MSBuildProjectDirectory), &quot;*.*proj&quot;)) != '')">false</EnableDefaultItems>
```

Comment:
These calls return arrays, so checking `Length != 0` might be more clear here, and wouldn't require MSBuild to coerce the return to a string for checking.

@rainersigwald one thing I think about with IO-based property functions - do these go through MSBuild's IO abstractions, or are they direct BCL calls? Are there things we need to worry about in general with the IO calls in MSBuild logic with multithreaded MSBuild (I know we want to prevent the use of some of these calls _in Tasks_)?

Reply:
I know we have code to not force an item expansion to stringify when compared to empty string, but I don't know if it works in this property-function case.

I think today we do straight passthrough to BCL but I wouldn't want to stop us from doing something better in the future.

But evaluation is already parallel in many cases and calling this should be fine (as long as you don't like depend on mutating filesystem state between projects).